### PR TITLE
Add FP exception handling, x86 specific

### PR DESCRIPTION
**Pull request by jacereda**
This would need some #ifdef work on other platforms.

### CODE REVIEW

File: core/fpe.h [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <fenv.h>
+
+static void enable_exceptions(int e) {
+    fenv_t fe = *FE_DFL_ENV;
+    fe.__control &= ~e;
+    fe.__mxcsr &= ~(e << 7);
+    fesetenv(&fe);
+}
```
Comment (bjorn3) [inline core/fpe.h]: This works only for x86, right?

Comment (jacereda) [reply core/fpe.h]: Yes, x86/x64 AFAIK.

**Comment by eriksvedang** [comment]
This is cool! There is a problem on macOS though, because of case insensitivity. The test file `/test-for-exceptions/fpe.carp` loads itself because it's called the same thing as the core library file. Please change the name of the test file to something more descriptive.

**Comment by eriksvedang** [comment]
I tried renaming the test from `fpe.carp` to `fpe_test.carp` but I still can't make it pass.

**Comment by hellerve** [comment]
I also can’t trivially get this to work. My errors do not align with what is expected.

We should probably revisit this at a later point, because I quite like the idea of this PR.

**Comment by eriksvedang** [comment]
Works on my Mac now, so that's great!

Fails in Windows, though the tests are not run there? So not sure what that is about...

**Comment by hellerve** [comment]
> Fails in Windows, though the tests are not run there? So not sure what that is about...

I’m not quite sure what you mean by that, the output seems to be from that test:

```
./test-for-exceptions/test-fpe.carp
In file included from out\main.c:32:
D:/a/Carp/Carp/core\fpe.h(7,8): error: no member named '__control' in 'struct fenv_t'
    fe.__control &= ~e;
    ~~ ^
D:/a/Carp/Carp/core\fpe.h(9,8): error: no member named '__mxcsr' in 'struct fenv_t'
    fe.__mxcsr &= ~(e << 7);
```

**Comment by eriksvedang** [comment]
Oh, OK. I forgot that the Windows CI also use the bash script now, sorry.

**Comment by eriksvedang** [comment]
I guess we need a workaround for that then...

**Comment by jacereda** [comment]
There must be some way to do this in a more portable way, but I still have to figure it out.

**Comment by eriksvedang** [comment]
Will close for now, please open again when if you get it to work on all platforms!

**Pull request closed without merging**